ens/ens.py

```diff
@@ -112,13 +112,19 @@ def from_web3(cls, w3: "Web3", addr: ChecksumAddress = None) -> "ENS":
         """
         Generate an ENS instance with web3
 
-        :param `web3.Web3` w3: to infer connection information
+        :param `web3.Web3` w3: to infer connection, middleware, and codec information
         :param hex-string addr: the address of the ENS registry on-chain. If not
             provided, defaults to the mainnet ENS registry address.
         """
         provider = w3.manager.provider
         middlewares = w3.middleware_onion.middlewares
-        return cls(cast("BaseProvider", provider), addr=addr, middlewares=middlewares)
+        ns = cls(cast("BaseProvider", provider), addr=addr, middlewares=middlewares)
+
+        # set codec and strict bytes type check flag from w3
+        ns.w3.strict_bytes_type_checking = w3.strict_bytes_type_checking
+        ns.w3.codec = w3.codec
```

Comment:
Ah, yep this is redundant... good eye 👍🏼

Reply:
> maybe some documentation around the codec and how to use it would be warranted in a future PR?

Yeah not sure here... We don't seem to have a lot of docs around it currently. I could go either way with that. I assume most users want to inherit the proper encoding / decoding that is native with the library.

Reply:
Having seen your top comment, I think I see this issue a bit clearer. Should we instead expose a `ENS.enable_strict_bytes_checking` flag that is a direct object reference to `ns.w3.enable_strict_bytes_checking`? That way it's less obscured on the standalone ENS instance? I could add that in and add documentation for it.